src/metrics/mongoose_metrics_hooks.erl

```diff
@@ -153,89 +154,99 @@ xmpp_send_element(Acc, _El) ->
 
 %% Roster
 
--spec roster_get(list(), jid:jid()) -> list().
-roster_get(Acc, #jid{lserver = LServer}) ->
-    mongoose_metrics:update(LServer, modRosterGets, 1),
+-spec roster_get(mongoose_acc:t(), jid:jid()) -> mongoose_acc:t().
+roster_get(Acc, _) ->
+    HostType = mongoose_acc:host_type(Acc),
+    mongoose_metrics:update(HostType, modRosterGets, 1),
     Acc.
 
--spec roster_set(Acc :: map(), JID :: jid:jid(), tuple(), tuple()) ->
-    metrics_notify_return().
-roster_set(Acc, #jid{server = Server}, _, _) ->
-    mongoose_metrics:update(Server, modRosterSets, 1),
+-spec roster_set(Acc :: any(), JID :: jid:jid(), tuple(), tuple()) ->
+    any().
+roster_set(Acc, #jid{lserver = LServer}, _, _) ->
+    {ok, HostType} = mongoose_domain_api:get_host_type(LServer),
```

Comment:
These are so inconsistent... ensuring host types in non-global hook Acc's seems like a good solution as @DenysGonchar suggested. No need to address it in this PR of course.

Reply:
True, but these is one of the hooks in which the Acc is `ok`. There are a few more similar cases in `mongoose_metrics_hooks` sadly.